### PR TITLE
Add rules command

### LINKED ISSUE #31: Add `/rules` command
Add a `/rules` command to display the rules of a subreddit.

This data can be pulled from the Reddit API. Unfortunately this means we have to add an API wrapper to the dependencies and create a Reddit app for the bot.

However, this command is useful when people ask if a post is allowed on a sub and you don't want to open Reddit to check their rules.

**Options**:
- `subreddit`: The subreddit to get the rules for.

### PR DESCRIPTION
**Pull request by TimJentzsch**
Relevant issue: Closes #31.

## Description:

Adds a `/rules` command to get the rules of a given subreddit. This will be useful for moderation when someone asks if a post is allowed on a subreddit. Instead of openening Reddit and looking for the sub you can just do it in Discord.

Unfortunately this also adds praw as dependency and adds another bot token, but I think it's worth it (and we might want to add other Reddit features along the line).

## Screenshots:

The response of the bot to the `/rules` command

## Checklist:

- [x] Code Quality
- [x] Pep-8
- [ ] Tests (if applicable)
- [x] Success Criteria Met
- [x] Inline Documentation
- [ ] Wiki Documentation (if applicable)

### CODE REVIEW

**Comment by MaxVanDeursen** [comment]
Sweet idea to include this command!

File: buttercup/cogs/rules.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        self.bot = bot
+        self.reddit_api = reddit_api
+
+    @cog_ext.cog_slash(
+        name="rules",
+        description="Get the rules of the specified subreddit.",
+        options=[
+            create_option(
+                name="subreddit",
+                description="The subreddit to get the rules of.",
```
Comment (MaxVanDeursen) [inline buttercup/cogs/rules.py:37]: It might be worth to mention here that the `/r/` prefix is not to be included

Comment (TimJentzsch) [reply buttercup/cogs/rules.py:37]: It should work with `/r/` as well! (I think) 😆

Comment (TimJentzsch) [reply buttercup/cogs/rules.py:37]: I just tested again, and it works

**Review by MaxVanDeursen — APPROVED** [review]

**Pull request merged**